### ISSUE TITLE
[BUG]: Attach Control not working in web-form

### ISSUE THREAD

**Issue by mohsinalimat** [kid1194/frappe-better-attach-control#5]
When I assign options {"allowed_file_types": ["image/*"]} in Attach Field Type then I will not work on web-form

in version-13 and version-14 as well.

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot buddy for reporting this..

I knew that there is an issue with the web form but I didn't find a way to fix it..

Can you please check the console for any error..
And please also check if only the `allowed_file_types` option is what makes the control stop working or other options cause that too..

I'm very thankful for your help..

Best regards..

**Comment by mohsinalimat** [comment]
in inspect element html we put **accept="image/*"** then only images option comes in file browser.

**Comment by mohsinalimat** [comment]
If I change below line in core code then after it works in web-form and doctype form.

I also check in version-13 and version-14.

![Screenshot 2023-01-31 at 1 33 18 PM](https://user-images.githubusercontent.com/973676/215701622-b4d5a4c3-06c9-4a40-9c46-6559ebb052c6.png)

**Comment by kid1194** [comment]
@mohsinalimat That's weird..

Anyway, I have updated the plugin, hopefully it will fix this issue..
Please give it a try and let me know if you find any bug..

Best regards..

**Comment by mohsinalimat** [comment]
Still not working in web-from in version-14.

**Comment by kid1194** [comment]
@mohsinalimat I couldn't find a solution for this problem..

I think that the problem isn't from the plugin, but from the `web_script.js` file..

Check this #4 for more info..

Best regards..

**Comment by kid1194** [comment]
@mohsinalimat Hey buddy..

I think that I have found a solution for the web form. Can you please update the plugin, move the config from the `Options` property to the `Better Attach Optiins` property and then check if the plugin works properly in desk and in web form..

To show gratitude for your contribution to this plugin, I have added your name in the README file..

Best regards..

**Comment by mohsinalimat** [comment]
Still same error.

**Comment by mohsinalimat** [comment]
> @mohsinalimat Hey buddy..
> 
> 
> 
> I think that I have found a solution for the web form. Can you please update the plugin, move the config from the `Options` property to the `Better Attach Optiins` property and then check if the plugin works properly in desk and in web form..
> 
> 
> 
> To show gratitude for your contribution to this plugin, I have added your name in the README file..
> 
> 
> 
> Best regards..

Please share a demo video for web-form restrictions.

**Comment by kid1194** [comment]
@mohsinalimat Sorry buddy, I can't test it currently..

The issue was that in `web_script.js`, the web form fields of type `Attach` and `Attach Image` gets their options property set to null if the property value is not an object and since the plugin options is JSON that means it gets removed..

So, I added a custom field to hold the plugin options in order to bypass this action of `web_script.js`, and the plugin gets the options from the custom field..

Have you tried it after putting the options in the custom field?

If yes and it still doesn't work, please let me know..

Best regards..

**Comment by mohsinalimat** [comment]
I have child table in web-form. And in child table assign the image property.

**Comment by kid1194** [comment]
Is there a field property called `Better Attach Options`?
It should appear under the `Options` property..

If yes, did you put the plugin options inside it instead of the `Options` property?

if yes, then did it work for you in web form or not?

**Comment by kid1194** [comment]
@mohsinalimat Hey buddy..

I just wanted to remind you about the questions I wrote in the above comment..

I'm still waiting for your reply..

Best regards..

**Comment by mohsinalimat** [comment]
> Is there a field property called `Better Attach Options`? It should appear under the `Options` property..
> 
> If yes, did you put the plugin options inside it instead of the `Options` property?
> 
> if yes, then did it work for you in web form or not?

I can not find any `Better Attach Options` property.

**Comment by kid1194** [comment]
@mohsinalimat There should be a field for the plugin options..

What version you are using? 
Are you using the Beta version? 

The field exists in the Beta-8 version...

**Comment by mohsinalimat** [comment]
> What version you are using? 

version-14 & version-13


> Are you using the Beta version? 

Yes, Beta-8

**Comment by mohsinalimat** [comment]
Web From

![Screenshot 2023-02-22 at 11 29 50 AM](https://user-images.githubusercontent.com/973676/220535936-d104c3f8-ddaa-4fdb-ab76-497791ba2a43.png)

Child Doctype

![Screenshot 2023-02-22 at 11 44 04 AM](https://user-images.githubusercontent.com/973676/220538169-4ebd0fe1-41a5-4e26-838c-0a12e5e61356.png)

**Comment by kid1194** [comment]
@mohsinalimat I believe that you understood me wrong..

The plugin field will only appear if the field type is `Attach` or `Attach Image`...

So, when you edit your attach field you will see it and not when editing the child table...

If you edit your `WF Attachment Child` and you edit the attach field, you will be able to see the plugin options field below the normal options field..

**Comment by mohsinalimat** [comment]
I can not find anything related the plugin options.

![Screenshot 2023-02-23 at 1 21 42 PM](https://user-images.githubusercontent.com/973676/220849612-cdc8588a-68db-4bf8-95d9-238a20fb9b73.png)

**Comment by kid1194** [comment]
@mohsinalimat The plugin setup code might not have created the custom field..

Can you please do the following..

1. Go to Home > Customization > Form Customization > Custom Field
2. In the list, check if there is a custom field called **Better Attach Options**
3. If there is no custom field, then create a new field and set the following data:
    - Document: `DocField`
    - Label: `Better Attach Options`
    - Field Type: `Small Text`
    - Field Name: `better_attach_options`
    - Depends On: `eval:in_list(['Attach', 'Attach Image'], doc.fieldtype)`
    - Insert Below: `options`

After creating the custom field, you will be able to see it when editing an Attach or Attach Image field..

If you can't find the document **DocField**, please let me know..

Best regards..

**Comment by mohsinalimat** [comment]
When I add **Better Attach Options** in Core Doctype of DocField then I will show. Issue is not solved in web form.

![Screenshot 2023-02-24 at 10 17 22 AM](https://user-images.githubusercontent.com/973676/221094173-8f08360f-378d-4def-b44e-df04a663f400.png)

**Comment by kid1194** [comment]
It should be working..

Anyway, I have made a small change to the plugin..

Please update and then when you open the web form, open the browser console and you will see the current options of the plugin. Check if the options from the console matches the options from the **Better Attach Options** field..

Also you will find in the console the uploader options that is being used when you click on the attach button and the file browser options when you click on the file browser button..

All the 3 options that you will find in the console should match with the options from the **Better Attach Options** field..

Thanks a lot buddy for your help..

**Comment by mohsinalimat** [comment]
```
		frappe._messages = {
 "Sr": "Sr"
};
		frappe.web_form_doc = {
 "allow_comments": 0,
 "allow_delete": 0,
 "allow_edit": 0,
 "allow_incomplete": 0,
 "allow_multiple": 0,
 "allow_print": 0,
 "apply_document_permissions": 0,
 "button_label": "Save",
 "creation": "2023-02-22 11:28:13.181937",
 "doc_type": "WF Attachment",
 "docstatus": 0,
 "doctype": "Web Form",
 "idx": 0,
 "in_edit_mode": false,
 "in_view_mode": false,
 "is_new": true,
 "is_standard": 0,
 "list_columns": [],
 "login_required": 0,
 "max_attachment_size": 0,
 "modified": "2023-02-22 11:28:15.646272",
 "modified_by": "Administrator",
 "module": "Website",
 "name": "wf-attachment",
 "owner": "Administrator",
 "published": 1,
 "route": "wf-attachment",
 "show_attachments": 0,
 "show_list": 0,
 "show_sidebar": 0,
 "title": "WF Attachment",
 "web_form_fields": [
  {
   "allow_read_on_all_link_options": 0,
   "creation": "2023-02-22 11:28:13.181937",
   "docstatus": 0,
   "doctype": "Web Form Field",
   "fieldname": "name1",
   "fieldtype": "Data",
   "hidden": 0,
   "idx": 1,
   "label": "Name",
   "max_length": 0,
   "max_value": 0,
   "modified": "2023-02-22 11:28:15.646272",
   "modified_by": "Administrator",
   "name": "192ee1f888",
   "owner": "Administrator",
   "parent": "WF Attachment",
   "parentfield": "web_form_fields",
   "parenttype": "Web Form",
   "read_only": 0,
   "reqd": 0,
   "show_in_filter": 0
  },
  {
   "allow_read_on_all_link_options": 0,
   "creation": "2023-02-22 11:28:13.181937",
   "docstatus": 0,
   "doctype": "Web Form Field",
   "fieldname": "wf_attachment_child",
   "fields": [
    {
     "fieldname": "name",
     "fieldtype": "Data",
     "label": "Name"
    },
    {
     "allow_bulk_edit": 0,
     "allow_in_quick_entry": 0,
     "allow_on_submit": 0,
     "better_attach_options": "{\"allowed_file_types\": [\"image/*\"]}",
     "bold": 0,
     "collapsible": 0,
     "collapsible_depends_on": null,
     "columns": 0,
     "creation": "2023-02-22 11:26:43.193107",
     "default": null,
     "depends_on": null,
     "description": null,
     "docstatus": 0,
     "doctype": "DocField",
     "documentation_url": null,
     "fetch_from": null,
     "fetch_if_empty": 0,
     "fieldname": "attach",
     "fieldtype": "Attach",
     "hidden": 0,
     "hide_border": 0,
     "hide_days": 0,
     "hide_seconds": 0,
     "idx": 1,
     "ignore_user_permissions": 0,
     "ignore_xss_filter": 0,
     "in_filter": 0,
     "in_global_search": 0,
     "in_list_view": 1,
     "in_preview": 0,
     "in_standard_filter": 0,
     "is_virtual": 0,
     "label": "Attach",
     "length": 0,
     "mandatory_depends_on": null,
     "max_height": null,
     "modified": "2023-02-24 13:15:05.238014",
     "modified_by": "Administrator",
     "name": "44e44f08b8",
     "no_copy": 0,
     "non_negative": 0,
     "oldfieldname": null,
     "oldfieldtype": null,
     "options": "{\"allowed_file_types\": [\"image/*\"]}",
     "owner": "Administrator",
     "parent": "WF Attachment Child",
     "parentfield": "fields",
     "parenttype": "DocType",
     "permlevel": 0,
     "precision": null,
     "print_hide": 0,
     "print_hide_if_no_value": 0,
     "print_width": null,
     "read_only": 0,
     "read_only_depends_on": null,
     "remember_last_selected_value": 0,
     "report_hide": 0,
     "reqd": 0,
     "search_index": 0,
     "set_only_once": 0,
     "show_dashboard": 0,
     "show_preview_popup": 0,
     "translatable": 0,
     "trigger": null,
     "unique": 0,
     "width": null
    }
   ],
   "fieldtype": "Table",
   "hidden": 0,
   "idx": 2,
   "label": "WF Attachment Child",
   "max_length": 0,
   "max_value": 0,
   "modified": "2023-02-22 11:28:15.646272",
   "modified_by": "Administrator",
   "name": "fb7fd3d135",
   "options": "WF Attachment Child",
   "owner": "Administrator",
   "parent": "WF Attachment",
   "parentfield": "web_form_fields",
   "parenttype": "Web Form",
   "read_only": 0,
   "reqd": 0,
   "show_in_filter": 0
  }
 ]
};
```

Update the custom app but still same issue.

**Comment by kid1194** [comment]
@mohsinalimat Can you please post a screenshot of the web browser console when using the web form..

**Comment by mohsinalimat** [comment]
Browser Console log of web form.

![Screenshot 2023-02-27 at 10 45 30 AM](https://user-images.githubusercontent.com/973676/221479952-8cee0fe2-f394-429b-b2d2-a91fadbf7524.png)

**Comment by kid1194** [comment]
@mohsinalimat I realized that the plugin doesn't override the default attach controls in web forms since it doesn't get loaded at all..

Injecting custom JavaScript and CSS files to web forms cannot be done unless specified in the hooks and for specific standard web forms only..

Because of that, I have made the plugin intercept the web form load and manually inject both JavaScript and CSS but I'm not sure that it will work since I don't know when the code will be executed, at the beginning of the web form load or at the end (which is the right time)..

So please update the plugin and check if it works automatically in web forms or not..

If it doesn't work, then the please try the following..

- If the web form is set as **Standard**, then the following code must be placed at the top of the exported web form js file..
- But if it is not set as **Standard**, then the desired web form must modified and following code must be placed at top inside the **Client Script** field of the web form..

**For v14**
```
frappe.require(['better_attach.bundle.js', 'better_attach.bundle.css']);
```

**For v13**
```
frappe.require([
    '/assets/frappe_better_attach_control/js/better_attach.js',
    '/assets/frappe_better_attach_control/css/better_attach.css',
]);
```

**For v12**
```
frappe.require([
    '/assets/frappe_better_attach_control/js/better_attach_v12.js',
    '/assets/frappe_better_attach_control/css/better_attach.css',
]);
```

**Comment by mohsinalimat** [comment]
Still facing same issue.

**Comment by kid1194** [comment]
@mohsinalimat Did you update the plugin?

If you did then please check frappe error log doctype and see if there is an entry from the plugin..

Have you tried the second method which is adding the js code?

**Comment by mohsinalimat** [comment]
I have update the app and uninstall and install again.


Second option is also tried but no luck.

**Comment by kid1194** [comment]
@mohsinalimat Did you check frappe error log doctype for any error entry posted by the plugin?

If you checked and there were no plugin errors then making the plugin work for web forms might not be possible..😔

I have checked frappe source code for any possible workaround but I couldn't find any other than what I mentioned in the comment before last..

The context injection hook seems to be executed before the web form which is why the injection code doesn't work..

Even if I inject the js and css files for every web page and form, web forms will override both js and css if the form is set as **Standard**..

**Comment by mohsinalimat** [comment]
For v14

frappe.require(['better_attach.bundle.js', 'better_attach.bundle.css']);


I used this code in web form script but error is said no file found.

**Comment by kid1194** [comment]
@mohsinalimat Then please try using the code for v13..

This might be our last hope with web form..

```
frappe.require([
    '/assets/frappe_better_attach_control/js/better_attach.js',
    '/assets/frappe_better_attach_control/css/better_attach.css',
]);
```

**Comment by kid1194** [comment]
@mohsinalimat I think that I have solved this problem..

Please update the plugin and then check if it works for web forms..

The plugin should load in web forms with any **Attach** or **Attach Image** fields..

If it doesn't then please check frappe error log for any entry posted by the plugin..

Thanks a lot buddy..

**Comment by mohsinalimat** [comment]
![Screenshot 2023-03-01 at 11 41 30 AM](https://user-images.githubusercontent.com/973676/222059239-1245f4fd-c502-4f66-a198-7f239b7b3a60.png)

**Comment by kid1194** [comment]
@mohsinalimat Finally we got something although it is an error..

I will fix it and get back to you..

Thanks a lot buddy..

**Comment by kid1194** [comment]
@mohsinalimat I have updated the plugin..

It should work but if the plugin was unable to load a specific file, instead of throwing an error like before it will post an error entry with the file name to the **Error Log** doctype..

Please give it another try..

Best regards..

**Comment by mohsinalimat** [comment]
> ![Screenshot 2023-03-01 at 11 41 30 AM](https://user-images.githubusercontent.com/973676/222059239-1245f4fd-c502-4f66-a198-7f239b7b3a60.png)

Still get same error.

**Comment by kid1194** [comment]
@mohsinalimat Thanks to you bro I have managed to load both js and css for v14..

v13 and v12 someone must test and see it it loads or not..

Anyway I have updated the plugin..

Please give it a try..

Best regards..

**Comment by mohsinalimat** [comment]
After take latest pull then below error comes.

![Screenshot 2023-03-03 at 12 59 33 PM](https://user-images.githubusercontent.com/973676/222658600-cb4794d5-f7a8-4fc7-b258-74fcef966c01.png)

**Comment by kid1194** [comment]
@mohsinalimat A million thanks for your contribution bro..

It looks like the js files must be built before being used directly so I built the js files manually for v14, v13 and v12..

Now it should work 😁

Please update then give it another try and let me know if you face any problem..

Best regards..

**Comment by mohsinalimat** [comment]
> Can you please do the following..
> 
> Go to Home > Customization > Form Customization > Custom Field
> In the list, check if there is a custom field called Better Attach Options
> If there is no custom field, then create a new field and set the following data:
> Document: DocField
> Label: Better Attach Options
> Field Type: Small Text
> Field Name: better_attach_options
> Depends On: eval:in_list(['Attach', 'Attach Image'], doc.fieldtype)
> Insert Below: options
> After creating the custom field, you will be able to see it when editing an Attach or Attach Image field..
> 
> If you can't find the document DocField, please let me know..



In this one issue is there. I can see in property list but I can not see in customize form. Then after I added in core doctype DocField doctype then it works.

**Comment by kid1194** [comment]
> In this one issue is there. I can see in property list but I can not see in customize form. Then after I added in core doctype DocField doctype then it works.

That is weird, because it's a custom field for **DocField** doctype so it must be included in any **DocField** doctype form..

Maybe it wasn't included because you didn't set the module for this custom field as **Frappe Better Attach Control**, which I forgot to tell you about..

**Comment by kid1194** [comment]
@mohsinalimat Hey bro..

Do you think that the plugin doesn't have any errors?
Do you think that it is production ready or not yet?

Best regards..

**Comment by mohsinalimat** [comment]
Finally Now, I can only pick image extension but After select image below error comes.

![Screenshot 2023-03-06 at 9 58 57 AM](https://user-images.githubusercontent.com/973676/223019683-93569c37-856f-4ebe-b8c9-1d3f88ee1201.png)

**Comment by kid1194** [comment]
Thanks bro for reporting this..

I have updated the plugin, please update and give it a try..

Best regards..

**Comment by mohsinalimat** [comment]
Please check this issue.

https://user-images.githubusercontent.com/973676/223749853-04b5589f-d7e7-4045-a27a-6be9dc7a4d21.mov

**Comment by kid1194** [comment]
@mohsinalimat I have checked the video and I think that the file success upload handler that is being used is of the original control..

I have updated the plugin and added a lot of console logs so if it doesn't work we can find out what is going wrong..

Please give it a try and get back to me..

Best regards..

**Comment by mohsinalimat** [comment]
Please check console log.

https://user-images.githubusercontent.com/973676/223918649-60c75410-9ec1-4788-aae8-868d93e13369.mov

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot buddy..

Now I know where is the problem..

**Comment by mohsinalimat** [comment]
Great 👍

**Comment by kid1194** [comment]
@mohsinalimat I have updated the plugin. It should work..

Can you please check if the same problem exists in desk?

Best regards..

**Comment by mohsinalimat** [comment]
When pick image and upload then memory goes high.

**Comment by kid1194** [comment]
@mohsinalimat Browser memory or server memory?

Can you please provide a screenshot..

**Comment by mohsinalimat** [comment]
Browser memory.

**Comment by kid1194** [comment]
@mohsinalimat please post a screenshot of the console and of the memory..

**Comment by mohsinalimat** [comment]
<img width="1552" alt="Screenshot 2023-03-09 at 8 38 29 PM" src="https://user-images.githubusercontent.com/973676/224066528-0a5bb28b-28a4-4c19-b9ee-47e9dd8e06e5.png">

<img width="1552" alt="Screenshot 2023-03-09 at 8 38 34 PM" src="https://user-images.githubusercontent.com/973676/224066569-fc189628-8de4-4e62-8c9d-7bc405f618d0.png">

<img width="1552" alt="Screenshot 2023-03-09 at 8 39 22 PM" src="https://user-images.githubusercontent.com/973676/224066661-bbf765ac-8e5a-42c5-86ce-8a564607e804.png">

**Comment by kid1194** [comment]
@mohsinalimat Thanks bro. It's a bug and I will fix it 😀..

**Comment by mohsinalimat** [comment]
<img width="1549" alt="Screenshot 2023-03-09 at 8 38 29 PM" src="https://user-images.githubusercontent.com/973676/224067212-1eb662d1-92cc-4abe-a478-4c3a176bb71f.png">

<img width="1536" alt="Screenshot 2023-03-09 at 8 38 34 PM" src="https://user-images.githubusercontent.com/973676/224067284-f6733254-9d6a-40eb-8317-63c35f19e0e5.png">

<img width="1437" alt="Screenshot 2023-03-09 at 8 41 45 PM" src="https://user-images.githubusercontent.com/973676/224067356-9e354f78-601b-4c04-9cf6-4aae68852273.png">

<img width="1440" alt="Screenshot 2023-03-09 at 8 42 53 PM" src="https://user-images.githubusercontent.com/973676/224067706-f4b7a404-ada8-4573-a3d7-b99c89326ad4.png">

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot bro for the screenshots..

There was a loop problem..

I have updated the plugin so please give it a try and get back to me..

Best regards..

**Comment by mohsinalimat** [comment]
Upload issue solve but image extension issue comes up. I already set `{"allowed_file_types": ["image/*"]}` in **Better Attach Options** and **Options**

**Comment by mohsinalimat** [comment]
Another issue in form attachment.

![Screenshot 2023-03-10 at 10 27 54 AM](https://user-images.githubusercontent.com/973676/224227310-7b8ea492-acf5-4470-ad15-a16caeb3699c.png)

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot bro..

I will fix everything and get back to you..

I was just waiting for the confirmation from you regarding the web form so I can fix it for desk too..

Best regards..

**Comment by mohsinalimat** [comment]
Image extension issue is there in web from also.

**Comment by kid1194** [comment]
@mohsinalimat Thanks a million buddy..

I have updated the plugin and fixed every known bug 😁

Please give it a try and check if there are still some bugs in both, web form and desk..

Best regards..

**Comment by mohsinalimat** [comment]
Now, upload the image but image extension is not working.

![Screenshot 2023-03-11 at 10 04 28 AM](https://user-images.githubusercontent.com/973676/224465071-352d0445-a5d0-4d8d-a520-5662137a5db2.png)


![Screenshot 2023-03-11 at 10 04 17 AM](https://user-images.githubusercontent.com/973676/224465069-dbb0b848-f257-4ae7-ac47-27d51d22b4af.png)

**Comment by kid1194** [comment]
@mohsinalimat My mistake bro..

I made some changes to the list of allowed files to reduce memory but it is supposed to be done after it get used in html..

I have fixed this issue and added support for use of regex in allowed file types by adding **$** as a prefix to the regex like `$image\/([a-z]+)`..

Please update and check if you still face any problem..

Many thanks to you bro for your help..

Best regards..

**Comment by mohsinalimat** [comment]
Still restrictions not working

**Comment by kid1194** [comment]
@mohsinalimat That is weird..

Does the allowed file types appear in the accept html attribute?

Anyway, I will check the code to see if I missed something..

Best regards..

**Comment by mohsinalimat** [comment]
I used in options {"allowed_file_types": ["image/*"]}



https://user-images.githubusercontent.com/973676/224465071-352d0445-a5d0-4d8d-a520-5662137a5db2.png

**Comment by kid1194** [comment]
@mohsinalimat I have updated the plugin and there will be some messages in the browser console..

Everything is supposed to work but please let me know if it doesn't and post a screenshot of the console..

Best regards..

**Comment by mohsinalimat** [comment]
Now, it's working but let me check tomorrow because take lots of browser memory in form and web form also.


I think log takes too much browser memory.

**Comment by mohsinalimat** [comment]
https://user-images.githubusercontent.com/973676/224614054-60577a7e-b8ff-4332-9f2d-ca059ee84d2f.mov

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot buddy for your help.. 

Just to clarify things, you have an attach field in a child table and because of the attach field, you can't edit a row or add a new one in that child table. Am I right?

What about if the attach field is not in a child table, does it work or not?

**Comment by mohsinalimat** [comment]
> What about if the attach field is not in a child table, does it work or not?

Not checked.

**Comment by mohsinalimat** [comment]
Yes. And it's sometimes this happened.

**Comment by kid1194** [comment]
@mohsinalimat I suspect that it is because of the dialog that is supposed to display the file when the filename is clicked..

Is the field type Attach or Attach Image?

Attach Image field doesn't have a dialog so can you please try the child table with Attach Image field..

And can you please try both Attach and Attach Image fields outside child table..

v2 will be dedicated to you bro for your enormous help 😁

**Comment by mohsinalimat** [comment]
It's Attach field type.

**Comment by kid1194** [comment]
@mohsinalimat I have changed a bit for v14 desk and web form controls in terms of the filename click.

I have put some console logs for the click event and removed the previous logs..

So please update and check the console after you click on the attached filename, edit button and add row button..

If you find any log in the console, please post a screenshot..

Best regards..

**Comment by mohsinalimat** [comment]
- In web form working perfect now.
 - Still Edit button not in form view, Please check below video.

https://user-images.githubusercontent.com/973676/224897444-0ba377a1-5554-42bc-9612-81863e118bd6.mp4

**Comment by kid1194** [comment]
@mohsinalimat I have updated the plugin..

I have disabled the uploaded files list and preview dialog that appears after clicking the file name, only if the field is placed inside a child table..

I'm trying to find out if the problem is from the dialog or not..

There should be a console log stating that the field is inside a child table but if it doesn't appear in console it means that the child table check code isn't working..

Please give it a try and then let me know if it is still working in web form and if the bug is fixed in desk..

Best regards..

**Comment by kid1194** [comment]
@mohsinalimat Hey buddy..

I just wanted to check if you have tried the last update or not.

Best regards..

**Comment by mohsinalimat** [comment]
I think all is good. And in form edit view there is bug in frappe framework.

**Comment by kid1194** [comment]
@mohsinalimat Thanks a lot bro..

So the problem was with the dialog of uploaded files..

I will create a custom dialog that is not bound to frappe dialogs, then I will update the plugin and let you know..

Can you please tell me about the frappe bug you found so I can check if I can help them fixing it..

Best regards..

**Comment by mohsinalimat** [comment]
In Attach field type in child table issue.

**Comment by kid1194** [comment]
@mohsinalimat I have updated the plugin with custom dialog for uploaded files..

Please give it a try..

> In Attach field type in child table issue.

You mean the edit dialog is still not opening?
Or it is something else?

**Comment by kid1194** [comment]
@mohsinalimat Bro, I just want to remind you about trying out the latest update..

Have you tried it?

> In Attach field type in child table issue.

You mean the edit dialog is still not opening?
Or it is something else?

**Comment by mohsinalimat** [comment]
> You mean the edit dialog is still not opening?
> Or it is something else?

Yes. Not opening

**Comment by kid1194** [comment]
@mohsinalimat Is the edit dialog not opening using the latest update? If yes, then I need to create a different way to display the list of uploaded files in child table..

Thanks a lot bro..

**Comment by mohsinalimat** [comment]
Yes, not opening in form view in child table.

**Comment by kid1194** [comment]
@mohsinalimat Brother, I have updated the plugin..

Now the child table dialog should open and everything should work..

Please give it a try and let me know if you find any bug or problem..

Please also try it with multiple files disabled, using Attach Image and also when editing an entry with already uploaded files, only if you have time..

Best regards..

**Comment by kid1194** [comment]
@mohsinalimat I just want to check if everything is working or there are any issues..

Best regards..

**Comment by mohsinalimat** [comment]
Still not opening in form view in child table.

**Comment by kid1194** [comment]
That's weird bro. I have made the dialog gets created on demand and gets destroyed when closed so it doesn't block the child table dialog. Just like dialog of frappe File Uploader..

Do you think that I should get rid of the dialog and find another way to list and preview the uploaded files?

Should I do that for child table only or for form also?

Thanks a lot for your help bro..

Best regards..

**Comment by kid1194** [comment]
@mohsinalimat Hey bro..

I have removed the dialog from the plugin and replaced it with a list that is displayed under the field..

I have removed the dialog from desk and web form.. 

So please whenever you are free, update the plugin and give it a try..

I hope that there will be no bugs this time..

Best regards..

**Comment by kid1194** [comment]
@mohsinalimat Hey bro..

I just want to check if everything is working well or there are still some bugs..

Best regards..

**Comment by nirajregmi** [comment]
> {"allowed_file_types": ["image/*"]}

I have update the plugin but still I am facing the issue.

**Comment by nirajregmi** [comment]
<img width="674" alt="image" src="https://github.com/kid1194/frappe-better-attach-control/assets/47034619/fb7b6df2-a966-4e5f-8614-b58141e5cc5e">
Please find here the screenshot. I have added the field but its not working doctype also. I am able to attach .xls file also.
<img width="352" alt="image" src="https://github.com/kid1194/frappe-better-attach-control/assets/47034619/86222368-58a9-49ef-a93f-421af20f8522">

**Comment by nirajregmi** [comment]
also in frappe standard webform i am not able to attach files. it is throwing exception 404 error

**Comment by nirajregmi** [comment]
@kid1194 please help me with this.

**Comment by kid1194** [comment]
@NirajRegmi Hey bro..

First, I would like to thank you for reporting problems and for your patience..

I have made a lot of changes hoping that it will work without any problem..

The custom field is no longer needed since the app now has a registered hook for website context where it checks if the page displayed is a web form or not..

For web forms, the app will get all the attach field options, build the js library if not already built and inject the field options to js..
If the web form doesn't have any attach field, then all this process will be skipped..

So please give it another try and let me know if you face any problem..

A million thanks to you bro, and I hope that putting a badge with your name on it in the README file will show some of my appreciation for you..

Best regards..
